src/hooks/Tokens.ts

```diff
@@ -179,8 +179,10 @@ export function useCurrency(currencyId: string | undefined): Currency | null | u
   const { chainId } = useActiveWeb3React()
   const isETH = currencyId?.toUpperCase() === 'ETH'
   const token = useToken(isETH ? undefined : currencyId)
-  const extendedEther = useMemo(() => (chainId ? ExtendedEther.onChain(chainId) : undefined), [chainId])
-  const weth = chainId ? WETH9_EXTENDED[chainId] : undefined
+  if (!chainId) return undefined
+
+  const ether = ExtendedEther.onChain(chainId)
```

Comment:
`ExtendedEther.onChain` creates a new object which causes dependencies of `useCurrency` to re-render unnecessarily. That's why we have a memoize extendedEther above.

Reply:
ty sir